app/src/main/kotlin/com/wire/android/ui/home/conversations/details/GroupConversationDetailsViewModel.kt

```diff
@@ -95,12 +96,12 @@ class GroupConversationDetailsViewModel @Inject constructor(
     }
 
     fun onGuestDialogDismiss() {
-        updateState(groupOptionsState.copy(isGuestUpdateDialogShown = false, isGuestAllowed = groupOptionsState.isGuestAllowed))
+        updateState(groupOptionsState.copy(isGuestUpdateDialogShown = false, isGuestAllowed = !groupOptionsState.isGuestAllowed))
```

Comment:
Do we need to update `isGuestAllowed` when the dialog is dismissed ?

Reply:
Yes, to avoid having the switch out of sync, the state is immediately inverted (without waiting for the response or dialog confirmation) and when the user dismisses the dialog the state is inverted again

i will attach a video of how it looks like